src/workers/_inputs.js

```diff
@@ -78,6 +91,48 @@ function loadMatrixMarketRaw(files) {
     }
 
     permuteGenes(cache.genes);
+
+    var annotations_file = files.filter(x => x.type == "annotations");
+    if (annotations_file.length == 1) {
+        var annotations_file = annotations_file[0]
+        var content = new Uint8Array(annotations_file.buffer);
+        var ext = annotations_file.name.split('.').pop();
+        let compressed = false;
+
+        if (ext == "gz") {
+            compressed = true;
+        }
+
+        let parsed = readCSVFromBuffer(content, compressed = compressed);
+
+        let diff = cache.matrix.numberOfColumns() - parsed.length;
+        // check if a header is present or not
+        let headerFlag = false;
+        if (diff === 0) {
+            headerFlag = false;
+        } else if (diff === -1) {
+            headerFlag = true;
+        } else {
+            throw "number of matrix cols is not equal to the number of cells in '" + annotations_file.name + "'";
+        }
+
+        let headers = [];
+        if (headerFlag) {
+            headers = parsed.shift();
+        } else {
+            parsed[0].forEach((x, i) => {
+                headers.push(`Column_${i + 1}`);
+            })
+        }
+
+        cache.annotations = {
```

Comment:
Shouldn't this be an object where the keys are the header names and the values are the column vectors? This would allow for easy extraction of any given column instead of looping over and pulling out the field of interest every time.